dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java

```diff
@@ -83,24 +84,45 @@ public String getFinalSqlExpression(String tableLoc) {
           result .append(" , ");
         }
         String [] rawExpandedTerm = splitIndex[i].split("\\.");
-        StringBuilder expandedTerm = formatExpandedTerm(rawExpandedTerm);
-        result.append(tableLoc).append(".jsonb").append(expandedTerm);
+        StringBuilder expandedTerm = formatExpandedTerm(tableLoc + ".jsonb",rawExpandedTerm);
+        result.append(expandedTerm);
       }
       result.append(")");
       return result.toString();
     }
 
   }
-  private StringBuilder formatExpandedTerm(String[] rawExpandedTerm) {
-    StringBuilder expandedTerm = new StringBuilder();
 
+  private StringBuilder formatExpandedTerm(String table, String[] rawExpandedTerm) {
```

Comment:
It is very difficult to understand this method. The two cases (with and witout [*]) are joined but should completely been split for readability. For example:

    void appendExpandedTerm(StringBuilder result, String tableLoc, String term) {
      if (term.contains("[*]") {
        appendExpandedArrayTerm(result, tableLoc, term);
      } else {
        appendExpandedSimpleTerm(result, tableLoc, term);
      }
    }